docs/developers/data/abi.md

```diff
@@ -279,3 +279,576 @@ Similarly, [enums](/developers/data/custom-types#custom-enums) are defined by:
+            "name": "getSum",
+            "mutability": "readonly",
+            "inputs": [],
+            "outputs": [
+                {
+                    "type": "BigUint"
+                }
+            ]
+        },
+        {
+            "docs": [
+                "Add desired amount to the storage variable."
+            ],
+            "name": "add",
+            "mutability": "mutable",
+            "inputs": [
+                {
+                    "name": "value",
+                    "type": "BigUint"
+                }
+            ],
+            "outputs": []
+        }
+    ],
+    "events": [],
+    "esdtAttributes": [
+        {
+            "ticker": "testBasic",
+            "type": "BigUint"
+        },
+        {
+            "ticker": "testVec",
+            "type": "List<u64>"
+        },
+        {
+            "ticker": "testEnum",
+            "type": "MyEnum"
+        },
+        {
+            "ticker": "testOption",
+            "type": "Option<TokenIdentifier>"
+        }
+    ],
+    "hasCallback": false,
+    "types": {
+        "MyEnum": {
+            "type": "enum",
+            "variants": [
+                {
+                    "name": "First",
+                    "discriminant": 0
+                },
+                {
+                    "name": "Second",
+                    "discriminant": 1
+                },
+                {
+                    "name": "Third",
+                    "discriminant": 2
+                }
+            ]
+        },
+    }
+}
+
+```
+
+Now, if we take a look into the folder structure of the contract, we should see the following updated folder structure containing the newly generated files in `output`:
+
+```
+adder
+├── output
+│   ├── adder.abi.json
+│   ├── adder.imports.json
+|   ├── adder.mxsc.json
+|   ├── adder.wasm
+│   ├── testBasic.esdt-abi.json
+│   ├── testEnum.esdt-abi.json
+│   ├── testOption.esdt-abi.json
+│   ├── testVec.esdt-abi.json
+```
+
+Each file contains the new struct with its name and the type field's description.
+
+__testEnum.esdt-abi.json:__
+```json
+{
+    "esdtAttribute": {
+        "ticker": "testEnum",
+        "type": "MyEnum"
+    },
+    "types": {
+        "MyEnum": {
+            "type": "enum",
+            "variants": [
+                {
+                    "name": "First",
+                    "discriminant": 0
+                },
+                {
+                    "name": "Second",
+                    "discriminant": 1
+                },
+                {
+                    "name": "Third",
+                    "discriminant": 2
+                }
+            ]
+        }
+    }
+}
+```
+
+__testOption.esdt-abi.json:__
+```json
+{
+    "esdtAttribute": {
+        "ticker": "testOption",
+        "type": "Option<TokenIdentifier>"
+    }
+}
+```
+
+__testVec.esdt-abi.json:__
+```json
+{
+    "esdtAttribute": {
+        "ticker": "testVec",
+        "type": "List<u64>"
+    }
+}
+```
+
+## Custom types
```

Comment:
Struct examples belong in the `Custom types: struct` section, not here

Reply:
Keep the examples as small as possible, this page can easily grow out of control.